library/bigip_vlan.py

```diff
@@ -154,300 +152,301 @@
 
-        if self.exists():
-            return True
-        else:
-            raise F5ModuleError("Failed to create the VLAN")
+        changes = self.changes.to_return()
+        result.update(**changes)
+        result.update(dict(changed=changed))
+        return result
+
+    def _set_changed_options(self):
+        changed = {}
+        for key in Parameters.returnables:
+            if getattr(self.want, key) is not None:
+                if key == 'tagged_interfaces' or key == 'untagged_interfaces':
+                    ifcs = self._parse_interfaces(key)
+                    changed['interfaces'] = ifcs
+                    self.want.interfaces = ifcs
 
-    def update(self):
-        changed = False
-        params = dict()
-        current = self.read()
-
-        check_mode = self.params['check_mode']
-        description = self.params['description']
-        name = self.params['name']
-        tag = self.params['tag']
-        partition = self.params['partition']
-        tagged_interfaces = self.params['tagged_interfaces']
-        untagged_interfaces = self.params['untagged_interfaces']
-
-        if untagged_interfaces is not None or tagged_interfaces is not None:
-            ifcs = self.api.tm.net.interfaces.get_collection()
-            ifcs = [str(x.name) for x in ifcs]
-
-            if len(ifcs) is 0:
-                raise F5ModuleError(
-                    'No interfaces were found'
-                )
-
-            pinterfaces = []
-            if untagged_interfaces:
-                interfaces = untagged_interfaces
-            elif tagged_interfaces:
-                interfaces = tagged_interfaces
-
-            for ifc in interfaces:
-                ifc = str(ifc)
-                if ifc in ifcs:
-                    pinterfaces.append(ifc)
-                else:
-                    raise F5ModuleError(
-                        'The specified interface "%s" was not found' % (ifc)
-                    )
-
-            if tagged_interfaces:
-                tmp = [dict(name=x, tagged=True) for x in pinterfaces]
-                if 'tagged_interfaces' in current:
-                    if pinterfaces != current['tagged_interfaces']:
-                        params['interfaces'] = tmp
                 else:
-                    params['interfaces'] = tmp
-            elif untagged_interfaces:
-                tmp = [dict(name=x, untagged=True) for x in pinterfaces]
-                if 'untagged_interfaces' in current:
-                    if pinterfaces != current['untagged_interfaces']:
-                        params['interfaces'] = tmp
+                    changed[key] = getattr(self.want, key)
+        if changed:
+            self.changes = Parameters(changed)
+
+    def _update_changed_options(self):
+        changed = {}
+        for key in Parameters.updatables:
+            if getattr(self.want, key) is not None:
+                if key == 'tagged_interfaces' or key == 'untagged_interfaces':
```

Comment:
this method is supposed to be a baseclass method. do not get in the habit of putting module specific tests here. if you're doing that, then those should go in the parameters class